### PR TITLE
Fixed an issue where merged metrics propositions were duplicated in the sendEvent result

### PR DESCRIPTION
**Pull request by XDex**
## Description

[TNT-43892](https://jira.corp.adobe.com/browse/TNT-43892) Fix Propositions with merged metrics being duplicated in sendEvent result

## Related Issue

[TNT-43892](https://jira.corp.adobe.com/browse/TNT-43892)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [x] I have made any necessary test changes and all tests pass.
- [x] I have run the Sandbox successfully.

### CODE REVIEW

**Comment by XDex** [comment]
@jonsnyder Right, let me add a functional test

**Comment by ninaceban** [comment]
> @jonsnyder Right, let me add a functional test

unit tests that assert the correct grouping and merging of the propositions.

**Review by ninaceban — COMMENTED** [review]
Andrei can you have a proposition with only `measurement` item schema?

**Comment by XDex** [comment]
@jonsnyder I've added a functional test, please take another look. Thank you!

**Review by ninaceban — CHANGES_REQUESTED** [review]
Could you please update the `groupDecisions` unit test cases to reflect this change?

File: test/functional/specs/Personalization/C28762.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  requestHooks: [networkLogger.edgeEndpointLogs]
+});
+
+test.meta({
+  ID: "C28762",
+  SEVERITY: "P0",
+  TEST_RUN: "Regression"
+});
+
+test("Test C28762: Merged metric propositions should be delivered", async () => {
```
Comment (jonsnyder) [inline test/functional/specs/Personalization/C28762.js:53]: You need to add a test to https://testrail.corp.adobe.com/index.php?/suites/view/25 and use that number here.

**Review by jonsnyder — APPROVED** [review]

**Comment by jfkhoury** [comment]
@XDex @ninaceban What's pending for this PR? Thanks!

**Comment by XDex** [comment]
> @XDex @ninaceban What's pending for this PR? Thanks!

@jfkhoury Sorry, I've been off on sick leave and CE rotation this past sprint and didn't get a chance to finish up with it. I just need to add a unit test and adjust the functional test number as Jon suggested, will address it soon.

File: test/functional/specs/Personalization/C28762.js (hunk 1 above)
Comment (XDex) [reply test/functional/specs/Personalization/C28762.js:53]: Hopefully @shammowla will provide me with access to TestRail today and I'll add it

**Comment by XDex** [comment]
@ninaceban I've added a unit test as requested, please take another look, thanks!

**Review by ninaceban — APPROVED** [review]
looks good to me

File: test/functional/specs/Personalization/C28762.js (hunk 1 above)
Comment (XDex) [reply test/functional/specs/Personalization/C28762.js]: @jonsnyder I finally got access to TestRail today and renamed both the default content and merged metrics functional tests, please take another look. Thanks!

**Comment by jonsnyder** [comment]
I'm looking at refactoring the personalization logic within the Web SDK and I came across this PR. This change does indeed eliminate duplicate propositions from the result object. However, it also prevents the default-content-items from rendering. Is this intentional?

**Comment by XDex** [comment]
@jonsnyder Yes, it's ok for default-content-items to not be rendered, since they don't contain any actual content. The propositions for default-content-items however **must** be included in display notifications, as we do need to record the fact that default-content items were displayed and reflect this in the reports.

**Pull request merged**